nntrainer/tensor/tensor.h

```diff
@@ -684,19 +684,23 @@ class Tensor {
                                   float beta = 0.0f) const;
 
   /**
-   * @brief     Transpose Tensor
-   * @param[in] direction to transpose ex) 0:2:1
-   * @retval    Calculated Tensor
+   * @brief Transpose Tensor
+   *
+   * @param direction to transpose ex) 0:2:1
+   * @param is_FC if layer == FC only transpose shape and clone data
+   * @return Tensor
    */
-  Tensor transpose(const std::string &direction) const;
+  Tensor transpose(const std::string &direction, bool is_FC = false) const;
```

Comment:
we do not need to add this is_FC. Tensor class must not know the type of Data.

Reply:
then how about add these function to tensor.cpp

```.cpp
Tensor &Tensor::transpose_shape(const std::string &direction, Tensor &out) const {

  NNTR_THROW_IF(!contiguous, std::invalid_argument)
    << getName() << " is not contiguous. Cannot transpose.";

  if (out.getData() == getData()) {
    Tensor tmp = clone();
    return tmp.transpose_shape(direction, out);
  }

  out.reshape(dim.transpose(direction));

  out.copyData(clone());

  return out;

}

Tensor Tensor::transpose_shape(const std::string &direction) const {
  Tensor result(dim);
  transpose_shape(direction, result);
  return result;
}
```

Reply:
```.py

    new_weights.push_back(*weights[0]);
    new_weights.push_back(*weights[1]);
    new_weights[0].reshape(new_weights[0].getDim().transpose("0:2:1"));
```